### PR TITLE
Unity 4.6 staging stv gc fixes

### PR DESCRIPTION
**Pull request by UnityAlex**

### CODE REVIEW

**Comment by UnityAlex** [comment]
'tana Link: 
http://katana.hq.unity3d.com/projects/Mono%20Runtime%20and%20Classlibs/builders?mono_branch=unity-4.6-staging-stv&boo_branch=unity-4.6-staging&unityscript_branch=unity-4.6-staging&cecil_branch=unity-4.6-staging&krait-signal-handler_branch=master&unity_branch=trunk

File: mono/metadata/object-internals.h [hunk 1]
```diff
@@ -228,6 +228,7 @@ typedef struct {
 	MonoObject  *async_callback;
 	MonoObject  *execution_context;
 	MonoObject  *original_context;
+	guint32      gchandle;
```
Comment (joncham) [inline mono/metadata/object-internals.h]: You can't just put this here. This represents a managed object that is allocated from the C# definition. When you use this field later to are writing to random memory.

**Comment by joncham** [comment]
This should not be merged as is. I left a note with the issue on the source.

**Comment by Tak** [comment]
Merged.

**Pull request merged**